### DISCUSSION TITLE
minimal python + blivet

### DISCUSSION THREAD

**Discussion by Geertsky** [storaged-project/blivet#1187] [Q&A]
Hello!
I'm looking for a method to create a python distribution that has the blivet module included. I've been trying different ways but haven't been successful. Does anyone have suggestions on what I could try?

Thanks!

**Comment by vojtechtrefny** [comment]
The python dependencies of blivet are pretty minimal, so that shouldn't be a problem. The biggest issue will be the C libraries which we (indirectly) depend on -- blivet is basically unusable without [libblockdev](https://github.com/storaged-project/libblockdev) (we use the python bindings via GObject Introspection) which is a C library with a lot of platform specific dependencies and we are also use libparted (which is also a C library) via the [pyparted](https://github.com/dcantrell/pyparted/) package. So I am not even sure if it would be possible to create such Python package/distribution.

**Reply by Geertsky** [reply]
OK, so then when I would include libblockdev + all dependencies and libbytesize + all dependencies, I would be good to go right?
For the python I've created a venv using pipenv with blivet pycairo pygobject and pyparted.

**Reply by vojtechtrefny** [reply]
Yes, I think so. You could also do what we do with our PyPI package -- we just tell the users they need to install all dependencies manually themselves. Not ideal, but packaging a Python package with C dependencies is never easy.
